commands/command_filter_process.go

```diff
@@ -99,6 +88,19 @@ func filterCommand(cmd *cobra.Command, args []string) {
 				n = ptr.Size
 			}
 		case "smudge":
+			closeOnce = new(sync.Once)
+			available = make(chan *tq.Transfer)
+
+			if supportsDelay {
+				q = tq.NewTransferQueue(
+					tq.Download,
+					getTransferManifestOperationRemote("download", cfg.Remote()),
+					cfg.Remote(),
+					tq.RemoteRef(currentRemoteRef()),
+				)
```

Comment:
If we use a new transfer queue on each smudge command, then we'll issue a whole new batch command for each object, which would be very inefficient, since we're essentially downloading each object one by one.  That's the issue that the filter-process is trying to avoid.

I think what we might want to do instead is track if the queue has been closed (say, by setting `q` to `nil` when we do), and then create a new one only if one doesn't exist.  Does that sound reasonable to you?

Reply:
Thank you for pointing me to right direction to solve the issue. I'll need a few days to create the appropriate solution and test it, will get back to you asap.

Reply:
I've pushed new commit: https://github.com/shalashik/git-lfs/commit/ddbe7eacab32d4ba83254c6d180f26afe3cfec69